pthread/pthread.c

```diff
@@ -260,6 +260,8 @@ int pthread_create(pthread_t *thread, const pthread_attr_t *attr,
 	ctx->cancelstate = PTHREAD_CANCEL_ENABLE;
 	*thread = (pthread_t)ctx;
 
+	mutexLock(pthread_common.pthread_list_lock);
```

Comment:
Deadlock can happen now - please use `_pthread_ctx_put` if error is returned from `beginthreadex`

Reply:
Note that `_pthread_ctx_put` unlocks the mutex